ghcide/session-loader/Development/IDE/Session.hs

```diff
@@ -479,12 +480,28 @@ loadSessionWithOptions recorder SessionLoadingOptions{..} dir = do
         -- files in the project so that `knownFiles` can learn about them and
         -- we can generate a complete module graph
     let extendKnownTargets newTargets = do
-          knownTargets <- forM newTargets $ \TargetDetails{..} ->
+          knownTargets <- concatForM  newTargets $ \TargetDetails{..} ->
```

Comment:
minor nit: Does this still need to be a concatForM? I think we can just use `forM` now and return a single element.

Reply:
For a single `TargetFile`, I am still generating multiple `TargetFile`'s, so I still need it.

 https://github.com/haskell/haskell-language-server/pull/3893/files#diff-d7140bbfca1b7aaafd8c82d1d6aa21a3f941eb27ba8f08e4d9fb13766b21b418R501

Reply:
ah ok then.